back/app/services/custom_field_service.rb

```diff
@@ -88,15 +88,15 @@ def generate_token(str)
     str.dup.concat('_', [*('a'..'z'), *('0'..'9')].sample(3).join)
   end
 
-  def cleanup_custom_field_values!(custom_field_values)
-    custom_field_values.each_key do |key|
-      value = custom_field_values[key]
-      is_boolean = !!value == value
-      next if is_boolean || value.present?
-
-      custom_field_values.delete key
+  # Removes all blank values from the values hash in place, except for `false` values,
+  # and returns self.
+  # @example
+  #  compact_custom_field_values!({a: 1, b: '', c: false, d: nil})
+  #  # => {a: 1, c: false}
+  def compact_custom_field_values!(cf_values)
```

Comment:
A matter of taste maybe but I'm not a big fan of abbreviations in variable names. Probably fairly clear in this instance I guess though.